runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/options/DataflowPipelineOptionsTest.java

```diff
@@ -177,9 +177,9 @@ public void testDefaultInvalidGcpTempLocation() {
   @Test
   public void testDefaultStagingLocationUnset() {
     DataflowPipelineOptions options = PipelineOptionsFactory.as(DataflowPipelineOptions.class);
+    options.setProject("");
     thrown.expect(IllegalArgumentException.class);
-    thrown.expectMessage(
-        "Error constructing default value for stagingLocation: gcpTempLocation is missing.");
+    thrown.expectMessage("--project is a required option.");
     options.getStagingLocation();
   }
```

Comment:
[one of the new tests should include applying BigQueryIO with validation enabled, which I think today will mimic the above.]